master/buildbot/test/unit/test_interpolate_secrets.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+from twisted.internet import defer
+from twisted.trial import unittest
+
+from buildbot.process.properties import Interpolate
+from buildbot.secrets.providers.base import SecretProviderBase
+from buildbot.secrets.secret import SecretDetails
+from buildbot.test.fake import fakemaster
+from buildbot.test.fake.fakebuild import FakeBuild
+from buildbot.test.util.config import ConfigErrorsMixin
+
+
+class FakeSecretStorage(SecretProviderBase):
+
+    def __init__(self, allsecretsInADict):
+        self.allsecrets = allsecretsInADict
+
+    def get(self, key):
+        if key in self.allsecrets:
+            return self.allsecrets[key], None
+        else:
+            return None, None
+
+
+class FakeBuildWithMaster(FakeBuild):
+
+    def __init__(self, master):
+        super(FakeBuildWithMaster, self).__init__()
+        self.master = master
+
+
+class FakeService(object):
+
+    def __init__(self, dictprop):
+        self.dict = dictprop
+
+    def get(self, key):
+        if key in self.dict:
+            return SecretDetails("FakeService", key, self.dict[key])
+        else:
+            return SecretDetails("FakeService", key, None)
+
+
+class FakeServiceManager(object):
+
+    def __init__(self, dictionary):
+        self.dictionary = dictionary
+
+    def __getitem__(self, dictionary):
+        return FakeService(self.dictionary)
+
+
+class TestInterpolateSecrets(unittest.TestCase, ConfigErrorsMixin):
+
+    def setUp(self):
+        self.master = fakemaster.make_master()
+        self.master.config.secretsManagers = [FakeSecretStorage({"foo": "bar",
+                                                                 "other": "value"})]
+        self.master.namedServices = FakeServiceManager({"foo": "bar",
```

Comment:
Why not use the real service manager? I think that a bit too much fakes here, makes me nervous.
Normally fakes are tested against real implementation to make sure they implement the same interface.

You need to do fakes when using real will make it too costly in term of runtime or test setup complexity.
I think this is not the case here.